tests/network/vmi_networking.go

```diff
@@ -899,16 +899,37 @@ var _ = SIGDescribe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:c
 
 				Expect(ping(podIP)).To(Succeed())
 
-				By("Restarting the vmi")
-				Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
-					&expect.BSnd{S: "sudo reboot\n"},
-					&expect.BExp{R: "reboot: Restarting system"},
-				}, 10)).To(Succeed(), "failed to restart the vmi")
-				tests.WaitUntilVMIReady(vmi, loginMethod)
-				if ipFamily == k8sv1.IPv6Protocol {
-					Expect(configureIpv6(vmi, api.DefaultVMIpv6CIDR)).To(Succeed(), "failed to configure ipv6 on vmi after restart")
+				By("Initiating DHCP server request after migration")
+
+				var dhClientDownCmd, dhClientUpCmd string
+				if ipFamily == k8sv1.IPv4Protocol {
+					dhClientDownCmd = "sudo cirros-dhcpc down eth0\n"
+					dhClientUpCmd = "sudo cirros-dhcpc up eth0\n"
+				} else {
+					dhClientDownCmd = "sudo dhclient -6 -r eth0\n"
+					dhClientUpCmd = "sudo dhclient -6 eth0\n"
 				}
+
+				Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+					&expect.BSnd{S: "\n"},
+					&expect.BExp{R: console.PromptExpression},
+					&expect.BSnd{S: dhClientDownCmd},
+					&expect.BExp{R: console.PromptExpression},
+					&expect.BSnd{S: tests.EchoLastReturnValue},
+					&expect.BExp{R: console.RetValue("0")},
+				}, 15)).To(Succeed(), "failed to release dhcp client")
+
+				Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+					&expect.BSnd{S: "\n"},
+					&expect.BExp{R: console.PromptExpression},
+					&expect.BSnd{S: dhClientUpCmd},
+					&expect.BExp{R: console.PromptExpression},
+					&expect.BSnd{S: tests.EchoLastReturnValue},
+					&expect.BExp{R: console.RetValue("0")},
+				}, 15)).To(Succeed(), "failed to rerun dhcp client")
```

Comment:
Please extract this out to a function, the test reader in this case is more interested in "what" is done, on "whom" and less on the "how".

It also feels awkward that the family is the condition for which DHCP client to use. It smells like the whole test needs to be restructured to make such additions easy to understand later.

Reply:
simplified the code